controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -296,44 +296,48 @@ func (r *ROSAControlPlaneReconciler) reconcileNormal(ctx context.Context, rosaSc
 		AWSCreator:     creator,
 	}
 
-	_, machineCIDR, err := net.ParseCIDR(rosaScope.ControlPlane.Spec.Network.MachineCIDR)
-	if err == nil {
-		spec.MachineCIDR = *machineCIDR
-	} else {
-		// TODO: expose in status
-		rosaScope.Error(err, "rosacontrolplane.spec.network.machineCIDR invalid", rosaScope.ControlPlane.Spec.Network.MachineCIDR)
-		return ctrl.Result{}, nil
-	}
+	if networkSpec := rosaScope.ControlPlane.Spec.Network; networkSpec != nil {
+		if networkSpec.MachineCIDR != "" {
+			_, machineCIDR, err := net.ParseCIDR(networkSpec.MachineCIDR)
+			if err != nil {
+				// TODO: expose in status
+				rosaScope.Error(err, "rosacontrolplane.spec.network.machineCIDR invalid", networkSpec.MachineCIDR)
+				return ctrl.Result{}, nil
+			}
+			ocmSpec.MachineCIDR = *machineCIDR
+		}
 
-	if rosaScope.ControlPlane.Spec.Network.PodCIDR != "" {
-		_, podCIDR, err := net.ParseCIDR(rosaScope.ControlPlane.Spec.Network.PodCIDR)
-		if err == nil {
-			spec.PodCIDR = *podCIDR
-		} else {
-			// TODO: expose in status.
-			rosaScope.Error(err, "rosacontrolplane.spec.network.podCIDR invalid", rosaScope.ControlPlane.Spec.Network.PodCIDR)
-			return ctrl.Result{}, nil
+		if networkSpec.PodCIDR != "" {
+			_, podCIDR, err := net.ParseCIDR(networkSpec.PodCIDR)
+			if err != nil {
+				// TODO: expose in status.
+				rosaScope.Error(err, "rosacontrolplane.spec.network.podCIDR invalid", networkSpec.PodCIDR)
+				return ctrl.Result{}, nil
+			}
+			ocmSpec.PodCIDR = *podCIDR
 		}
-	}
 
-	if rosaScope.ControlPlane.Spec.Network.ServiceCIDR != "" {
-		_, serviceCIDR, err := net.ParseCIDR(rosaScope.ControlPlane.Spec.Network.ServiceCIDR)
-		if err == nil {
-			spec.ServiceCIDR = *serviceCIDR
-		} else {
-			// TODO: expose in status.
-			rosaScope.Error(err, "rosacontrolplane.spec.network.serviceCIDR invalid", rosaScope.ControlPlane.Spec.Network.ServiceCIDR)
-			return ctrl.Result{}, nil
+		if networkSpec.ServiceCIDR != "" {
+			_, serviceCIDR, err := net.ParseCIDR(networkSpec.ServiceCIDR)
+			if err != nil {
+				// TODO: expose in status.
+				rosaScope.Error(err, "rosacontrolplane.spec.network.serviceCIDR invalid", networkSpec.ServiceCIDR)
+				return ctrl.Result{}, nil
+			}
+			ocmSpec.ServiceCIDR = *serviceCIDR
 		}
+
+		ocmSpec.HostPrefix = networkSpec.HostPrefix
+		ocmSpec.NetworkType = networkSpec.NetworkType
```

Comment:
outside networkSpec Do you need to set the type explicit to "OVNKubernete"  OR ocm set to "OVNKubernete" if its empty ?

Reply:
ocm sets "OVNKubernete" by default if not set